test/unit/specs/directives/internal/class_spec.js

```diff
@@ -53,5 +53,16 @@ describe(':class', function () {
     arr.push('f')
     dir.update(arr)
     expect(el.className).toBe('a f')
+    // test array with objects
+    dir.update(['x', {y: true, z: true}])
+    expect(el.className).toBe('a x y z')
+    dir.update(['x', {y: true, z: false}])
+    expect(el.className).toBe('a x y')
+    dir.update(['x', {z: true}])
```

Comment:
The test was passing previously with the bug in `contains`, so we need a bit more variation in the tests here - e.g. switching the order of the items in the array / changing `'x'` to something else.